windows/RNSVG/SvgView.cpp

```diff
@@ -189,16 +221,54 @@ void SvgView::Unload() {
   if (m_group) {
     m_group.Unload();
   }
-
-  if (m_canvas) {
-    m_canvas.RemoveFromVisualTree();
-    m_canvas = nullptr;
-  }
 }
 
-void SvgView::InvalidateCanvas() {
-  if (m_hasRendered) {
-    m_canvas.Invalidate();
+void SvgView::Invalidate() {
+  if (!m_loaded) {
+    return;
   }
+
+  m_brushes.Clear();
+  m_templates.Clear();
+
+  com_ptr<ID2D1DeviceContext> deviceContext{get_self<D2DDeviceContext>(DeviceContext())->Get()};
+
+  Size size{static_cast<float>(ActualWidth()), static_cast<float>(ActualHeight())};
+
+  xaml::Media::Imaging::SurfaceImageSource surfaceImageSource(
+      static_cast<int32_t>(size.Width), static_cast<int32_t>(size.Height));
+  com_ptr<ISurfaceImageSourceNativeWithD2D> sisNativeWithD2D{surfaceImageSource.as<ISurfaceImageSourceNativeWithD2D>()};
+
+  // Associate the Direct2D device with the SurfaceImageSource.
+  com_ptr<ID2D1Device> device{get_self<D2DDevice>(Device())->Get()};
+  sisNativeWithD2D->SetDevice(device.get());
+
+  com_ptr<IDXGISurface> dxgiSurface;
+
+  RECT updateRect{0, 0, static_cast<long>(size.Width), static_cast<long>(size.Height)};
```

Comment:
Oh the hilarity of yet another rectangle type in the same codebase. I mean, you're constrained by the APIs you're calling, so 🤷‍♂️.
This one had me do a double take because there might have been an error, given that `RECT` is LTRB not LTWH. But... since it's 0 offset it's the same.
For the record, that kind of mistake is why I think it's good to have explicit conversion functions for these. Example `RECTFromSize()`. There are so many casts and conversions in this code and so far I think they're right but also this stuff is error prone.